x/pricefeed/simulation/genesis.go

```diff
@@ -2,21 +2,96 @@ package simulation
 
 import (
 	"fmt"
+	"time"
 
 	"github.com/cosmos/cosmos-sdk/codec"
 	"github.com/cosmos/cosmos-sdk/types/module"
 
 	"github.com/kava-labs/kava/x/pricefeed/types"
+	pricefeed "github.com/kava-labs/kava/x/pricefeed/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
 // RandomizedGenState generates a random GenesisState for pricefeed
 func RandomizedGenState(simState *module.SimulationState) {
 
-	// TODO implement this fully
-	// - randomly generating the genesis params
-	// - overwriting with genesis provided to simulation
-	pricefeedGenesis := types.DefaultGenesisState()
+	// get the params with xrp, btc and bnb to usd
+	// define getPricefeedSimulationParams to return params with xrp:usd, btc:usd, bnb:usd
+	params := getPricefeedSimulationParams() // TODO QUESTION? IS THIS CORRECT? NEED TO CALL ANOTHER METHOD EG getPricefeedSimulationParams() ??
 
+	// now go through and modify the params adding the accounts as an oracle to each of the markets
+	genPrices := []types.PostedPrice{}
+	newMarkets := []types.Market{}
+	for _, market := range params.Markets {
+		oracles := []sdk.AccAddress{}
```

Comment:
After reviewing the application state that is exported after the sim, I think we should change the genesis state to only have one oralce - otherwise the prices don't get changed often enough. I would refactor like so:
```
params := getPricefeedSimulationParams()
	markets := []types.Market{}
	genPrices := []types.PostedPrice{}
	// chose one account to be the oracle
	oracle := simState.Accounts[simulation.RandIntBetween(simState.Rand, 0, len(simState.Accounts))]
	for _, market := range params.Markets {
		updatedMarket := types.Market{market.MarketID, market.BaseAsset, market.QuoteAsset, []sdk.AccAddress{oracle.Address}, true}
		markets = append(markets, updatedMarket)
		genPrice := types.PostedPrice{market.MarketID, oracle.Address, getInitialPrice(market.MarketID), simState.GenTimestamp.Add(time.Hour * 24)}
		genPrices = append(genPrices, genPrice)
	}
	params = types.NewParams(markets)
	pricefeedGenesis := types.NewGenesisState(params, genPrices)
```

Reply:
okk